application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -50,6 +55,11 @@
     private final TagSystem tagSystem;
     private final Predicate<String> hasRequiredRole;
 
+    /**
+     * stolen from {@link org.togetherjava.tjbot.routines.ModAuditLogRoutine}
+     */
+    private static final Color MOD_AUDIT_LOG_COLOR = Color.decode("#4FC3F7");
+
```

Comment:
> If you want the same color than the other class, make it package private and use it directly from `ModAuditLogWriter` then instead of duplicating it

What do you mean by that? I don't quite understand

Reply:
You are duplicating the code from the other class and by that introduce a soft dependency that will stale very soon, thats super fragile. For example, suppose the color of the `ModAuditLogRoutine` is changed to green. People will forget to adjust this constant here as well and then you have two different colors...

Instead, both classes should use the same constant and not 2 different. Make the constant accessible by making it package-private (remove the `private`) and then use it directly `ModAuditLogRoutine.AMBIENT_COLOR`.

Reply:
They are not in the same package

Reply:
My first comment is that you should move this into `ModAuditLogWriter` and that is in the same package.

Generally, the big design flaw in your PR is that you have a lot of the "mod audit log writing" logic in `TagCommand` but thats not its responsibility. Everything should be in `ModAuditLogWriter` instead, tightly boundled with the rest of the mod stuff.

Everything that `TagCommand` should do is just calling `ModAuditLogWriter.write("hello", new Attachment("foo.md", "bla"));` and thats it, no other changes to the class.

Reply:
yes, I have moved that. I am also working on merging the `log()` methods in `TagManageCommand` to one method, as you said in https://github.com/Together-Java/TJ-Bot/pull/296/files/4253473a98c315767a4f18e66986549c734b8538#r762454212, to mininize the amount of code in `TagManageCommand`

Reply:
> My first comment is that you should move this into `ModAuditLogWriter` and that is in the same package.

What do you mean by "and that is in the same package"? do you suggest to put `ModAuditLogWriter` and `ModAuditLogRoutine` in the same package?

Reply:
Either that or create some sort of utility class which both have access to. But yeah, why not put both into the same package.

Reply:
currently `ModAuditLogRoutine` is in `routines` package, do you want to move it to `moderation`?
btw both packages currently have only one class.

Reply:
Lets do it like that: There is this package that contains BanCommand, KickCommand. Afaik it has a ModerationUtils class or similar. Make this class public and move the ambient color to there.

Then keep ModAuditLogRoutine where it is and put ModAuditLogWriter next to it, in that routine package.

Reply:
`ModAuditLogWriter` doesn't belong at all to the `routines` package. this package is for routines, and as it says in the javadoc, (not mine) `Routines are actions that are executed periodically on a schedule.` I think should keep the classes where they are and move the color to `ModerationUtils`.

Reply:
Sounds like a plan.

Reply:
or, we could keep it the way it is to have more flexibility for changing the color to a different color

Reply:
Then it should be a different color right from the beginning in my opinion. Maybe a different shade of that color. Then I am good with it as well.

Reply:
ok, I'll make it a different shade. I think `#3788AC` is a good one

Reply:
done.